File: Medium/19. Remove Nth Node From End of List/solution.java
/**
 * Definition for singly-linked list. public class ListNode { int val; ListNode
 * next; ListNode(int x) { val = x; } }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if (head == null || head.next == null) {
            return null;
        }
        int size = 0;
        ListNode copy = head;
        while (copy != null) {
            copy = copy.next;
            size++;
        }
        copy = head;
        if (n == 1) {
            while (copy.next.next != null) {
                copy = copy.next;
            }
            copy.next = copy.next.next;
        } else {
            ListNode newhead = head;
            int i = 0;
            while (i < size - n) {
                newhead = newhead.next;
                i++;
            }
            newhead.val = newhead.next.val;
            newhead.next = newhead.next.next;
        }

        return head;
    }
}

File: Medium/19. Remove Nth Node From End of List/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        
        curr = head
        length: int = 0
        while curr:
            length += 1
            curr = curr.next
        
        target = length - n

        dummy = ListNode(-1)
        dummy.next = head
        curr = dummy

        for _ in range(target):
            curr = curr.next
        
        curr.next = curr.next.next
        return dummy.next